src/bin/registerRule.js

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Web3 = require('web3');
+const OpenST = require('@openstfoundation/openst.js');
+
+const connected = require('../connected');
+const ChainConfig = require('../config/chain_config');
+const logger = require('../logger');
+
+const { version } = require('../../package.json');
+
+program
+  .version(version)
+  .name('RegisterRule')
+  .arguments('<config> <ruleName> <ruleAddress> <ruleAbi>')
+  .description('An executable to register rules in tokenrules.')
+  .action(
+    async (config, ruleName, ruleAddress, ruleAbi) => {
+      await connected.run(
+        config,
+        async (chainConfig, connection) => {
+          const registerRuleTxOptions = {
+            from: connection.auxiliaryAccount.address, // MasterKey is Worker address
+            gasPrice: chainConfig.auxiliaryGasPrice,
+          };
+          const auxiliaryWeb3 = new Web3(connection.auxiliaryWeb3);
+          const tokenRules = new OpenST.ContractInteract.TokenRules(
+            auxiliaryWeb3,
+            chainConfig.openst.tokenRules,
+          );
+          await tokenRules.registerRule(ruleName, ruleAddress, ruleAbi, registerRuleTxOptions);
+          logger.info(`Rule ${ruleName} registered!`);
+          logger.info('Validating registered rule...');
+          const rule = await tokenRules.getRuleByName(ruleName);
```

Comment:
Do we need to make another call to verify rule is registered?  
Receipt status should also tell if the rule is registered. It could save one extra geth call.

Reply:
Do you see an optimization concern in JLP 😮 ?
The call also tests getByRuleName interact method. 
This has a little advantage of verifying with matching exact name and address 🤔  .

Reply:
Why not? It's extra code/call.